tests/analyse.py

```diff
@@ -0,0 +1,49 @@
+from tests import queries
+
+
+def analyze_mining(deployment,last_layer, layers_per_epoch, layer_avg_size, total_pods):
+    # need to filter out blocks that have come from last layer
+    blockmap, layermap = queries.get_blocks_per_node_and_layer(deployment)
+    queries.print_node_stats(blockmap)
+    queries.print_layer_stat(layermap)
+
+    xl = [len(layermap[str(x)]) for x in range(layers_per_epoch) if str(x) in layermap]
+    print(str(xl))
+
+    first_epoch_blocks = sum(xl)
+
+    # count all blocks arrived in relevant layers
+    total_blocks = sum([len(layermap[str(x)]) for x in range(last_layer) if str(x) in layermap])
+
+    atxmap = queries.get_atx_per_node(deployment)
+    total_atxs = sum([len(atxmap[x]) for x in atxmap])
+
+    print("atx created " + str(total_atxs))
+    print("blocks created " + str(total_blocks) + " first epoch blocks " + str(first_epoch_blocks))
+
+    genesis_block_deviation = (total_pods - layer_avg_size) * 2 * layers_per_epoch
+
+    if genesis_block_deviation > 0:
+        print("subtracting deviation of " + str(genesis_block_deviation) + " blocks for block counting")
+        total_blocks -= genesis_block_deviation
+
+    assert queries.get_nodes_up(deployment) == total_pods
+    assert total_pods == len(blockmap)
+    # remove blocks created in first epoch since first epoch starts with layer 1
+    assert ((total_blocks - first_epoch_blocks) / (last_layer - layers_per_epoch)) / layer_avg_size == 1
+    assert total_atxs == int((last_layer / layers_per_epoch)) * total_pods
+
+    # assert that a node has created one atx per epoch
+    for node in atxmap:
+        mp = set()
+        for blk in atxmap[node]:
+            mp.add(blk[4])
+        assert len(atxmap[node]) / int((last_layer / layers_per_epoch) + 0.5) == 1
+        if len(mp) != int((last_layer / layers_per_epoch)):
+            print("mp " + ','.join(mp) + " node " + node + " atxmap " + str(atxmap[node]))
+        assert len(mp) == int((last_layer / layers_per_epoch))
+
+    # assert that each node has created layer_avg/number_of_nodes
+    for node in blockmap:
+        first_epoch_blocks = sum([len(blockmap[node][1][str(x)]) for x in range(layers_per_epoch) if str(x) in blockmap[node][1]])
```

Comment:
it will be probably easier to understand if you'll break down this and below line to a few more lines. (giving meaningful names to params). before asserting